cmd/utils.go

```
@@ -0,0 +1,54 @@
+package cmd
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	psaApi "k8s.io/pod-security-admission/api"
+)
+
+func IgnoreNamespaceSelector(field string) string {
+	ignoredNamespaces := []string{"kube-system", "kube-public", "kube-node-lease"}
+	selectors := make([]fields.Selector, 0)
+	for _, n := range ignoredNamespaces {
+		selectors = append(selectors, fields.OneTermNotEqualSelector(field, n))
+	}
+	return fields.AndSelectors(selectors...).String()
+}
+
+func GetPods() *v1.PodList {
+	listOptions := metav1.ListOptions{FieldSelector: IgnoreNamespaceSelector("metadata.namespace")}
+	pods, err := clientset.CoreV1().Pods("").List(context.TODO(), listOptions)
+	if err != nil {
+		panic(err.Error())
+	}
+	return pods
+}
+
+func GetPodsByNamespace(namespace string) *v1.PodList {
+	listOptions := metav1.ListOptions{}
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), listOptions)
+	if err != nil {
+		panic(err.Error())
```

Probably better to return the error. Then the migration loop can just log the error and continue in a best-effort fashion (i.e. skip the failed namespace)

---

Done